### PR TITLE
fix compiler error: #else without #if

### PR DESCRIPTION
**Pull request by meta-space**
Changes introduced in AsyncTCP 2.1.0 lead to breakage in downstream projects: https://github.com/esphome/issues/issues/5258 

This PR is pretty quick and dirty. I only removed the problematic preprocessor directives without doing any further sanity checks

### CODE REVIEW

File: src/AsyncTCP.cpp [hunk 1]
```diff
@@ -1017,7 +1018,6 @@ int8_t AsyncClient::_poll(tcp_pcb* pcb){
 }
 
 void AsyncClient::_dns_found(struct ip_addr *ipaddr){
-#if LWIP_IPV4 && LWIP_IPV6
```
Comment (PruteanuVlad) [inline src/AsyncTCP.cpp]: I don't think that this is necessary.

File: src/AsyncTCP.cpp [hunk 2]
```diff
@@ -1017,7 +1018,6 @@ int8_t AsyncClient::_poll(tcp_pcb* pcb){
...
     if(ipaddr && ipaddr->u_addr.ip4.addr){
         connect(IPAddress(ipaddr->u_addr.ip4.addr), _connect_port);
```
Comment (PruteanuVlad) [inline src/AsyncTCP.cpp]: Add
```
#else
    if (ipaddr && ipaddr->addr)
    {
        connect(IPAddress(ipaddr->addr), _connect_port);
#endif

**Review by PruteanuVlad — CHANGES_REQUESTED** [review]
I took a look at the previous version of the code and I think that you should make these changes.

File: src/AsyncTCP.cpp (hunk 2 above)
Comment (meta-space) [reply src/AsyncTCP.cpp]: Thanks for taking the time to review the PR. I was mainly trying to raise awareness :)
I also noticed that another PR which tries to fix the same issue is also pending

**Comment by jwilleke** [comment]
Why is this still an issue?
It has been weeks.

**Comment by npon97** [comment]
Experiencing the same issue. Looks like line 1020 should be on line 752

**Comment by meta-space** [comment]
> Why is this still an issue? It has been weeks.

I hope maintainer(s) are doing ok. I've downgraded the AsyncTCP dependency in my project Version to 2.0.1 as a temporary workaround.

**Comment by meta-space** [comment]
I closed this PR; 
as far as I can see, the proposed changes are already part of AsyncTCP Release 2.1.3

**Pull request closed without merging**